--- ../src-base/minecraft/net/minecraft/block/BlockFarmland.java
+++ ../src-work/minecraft/net/minecraft/block/BlockFarmland.java
@@ -1,6 +1,8 @@
 package net.minecraft.block;
 
 import java.util.Random;
+
+import carpet.CarpetSettings;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyInteger;
@@ -81,11 +83,27 @@
     {
         IBlockState iblockstate = Blocks.field_150346_d.func_176223_P();
         p_190970_1_.func_175656_a(p_190970_2_, iblockstate);
-        AxisAlignedBB axisalignedbb = iblockstate.func_185890_d(p_190970_1_, p_190970_2_).func_186670_a(p_190970_2_);
+        AxisAlignedBB axisalignedbb;
+        if (!CarpetSettings.farmlandBug)
+        {
+            axisalignedbb = iblockstate.func_185890_d(p_190970_1_, p_190970_2_).func_186670_a(p_190970_2_);
 
+        }
+        else
+        {
+            axisalignedbb = field_185665_b.func_186670_a(p_190970_2_);
+        }
+
         for (Entity entity : p_190970_1_.func_72839_b((Entity)null, axisalignedbb))
         {
-            entity.func_70107_b(entity.field_70165_t, axisalignedbb.field_72337_e, entity.field_70161_v);
+            double d0 = Math.min(axisalignedbb.field_72337_e - axisalignedbb.field_72338_b, axisalignedbb.field_72337_e - entity.func_174813_aQ().field_72338_b);
+            if (!CarpetSettings.farmlandBug)
+            {
+                entity.func_70634_a(entity.field_70165_t, entity.field_70163_u + d0 + 0.001D, entity.field_70161_v);
+            } else
+            {
+                entity.func_70634_a(entity.field_70165_t, axisalignedbb.field_72337_e, entity.field_70161_v);
+            }
         }
     }
 
